### PR TITLE
Add prior diagnostics, and failsafe MSE variance diagnostics.

### PR DESCRIPTION
**Pull request by ilopezgp**
Adds:

- Prior diagnostics in both constrained and unconstrained space.
- NaN-safe diagnostics of the MSE ensemble variance both for ensemble and unscented (i.e., quadrature) methods.

In order to add these diagnostics:
- Adds `construct_mean` and `construct_cov` UKI methods that take vectors, not only matrices. These are useful for diagnostic purposes, and should be ported to the upstream package EKP.jl.

### CODE REVIEW

File: src/Diagnostics.jl [hunk 1]
```diff
@@ -141,6 +142,59 @@ function io_dictionary_val_reference(
...
+    ϕ_upp_std = transform_unconstrained_to_constrained(priors, u_mean .+ 1.0)
+
+    io_dict = Dict(
+        "u_mean_prior" => Base.setindex(orig_dict["u_mean_prior"], u_mean, :field),
+        "phi_mean_prior" => Base.setindex(orig_dict["phi_mean_prior"], ϕ_mean, :field),
+        "u_var_prior" => Base.setindex(orig_dict["u_var_prior"], u_var, :field),
+        "phi_low_unc_prior" => Base.setindex(orig_dict["phi_low_unc_prior"], ϕ_low, :field),
+        "phi_upp_unc_prior" => Base.setindex(orig_dict["phi_upp_unc_prior"], ϕ_upp, :field),
+        "phi_low_std_prior" => Base.setindex(orig_dict["phi_low_std_prior"], ϕ_low_std, :field),
+        "phi_upp_std_prior" => Base.setindex(orig_dict["phi_upp_std_prior"], ϕ_upp_std, :field),
```
Comment (charleskawczynski) [inline src/Diagnostics.jl:193]: This is probably not a good pattern to follow, we should probably be setting these fields in a vectorized way

Comment (ilopezgp) [reply src/Diagnostics.jl:193]: I see, how would that look?

Comment (charleskawczynski) [reply src/Diagnostics.jl:193]: I'd need to see the context of how this is used, but it basically boils down to avoiding calling `Base.setindex`

**Pull request merged**